src/solvers/iterators.jl

```diff
@@ -0,0 +1,104 @@
+#
+# SweepIterator
+#
+
+mutable struct SweepIterator
+  sweep_kws
+  region_iter
+  which_sweep::Int
+end
```

Comment:
What about parameterizing this  struct by the types of the fields `sweep_kws` and `region_iter`? Is it meant to be dynamic, i.e. is it expected that those types might change?